### PR TITLE
Propagate all warnings captured in coroutine test functions to pytest

### LINKED ISSUE #4656: aiohttp.pytest_plugin suppresses all the warnings
🐞 **Describe the bug**

`aiohttp.pytest_plugin` suppresses warnings issued from test. It looks like _runtime_warning_context() catches all the warnings so no warning reaches pytest.warnings plugin.

💡 **To Reproduce**

Simple testcase that shows how pytest works with warnings. It would print warning for test_foo only.

```python
import warnings

def test_foo():
    warnings.warn('sync warning')


async def test_afoo():
    warnings.warn('async warning')
```

💡 **Expected behavior**

pytest catches warning.

📋 **Logs/tracebacks**

```python-traceback (paste your traceback in the next line)
========================= test session starts ==========================
platform linux -- Python 3.7.3, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- python3.7
cachedir: .pytest_cache
rootdir: ..., inifile: pytest.ini
plugins: aiohttp-0.3.0
collected 2 items                                                      

tests/test_foo.py::test_foo PASSED
tests/test_foo.py::test_afoo 
PASSED

=========================== warnings summary ===========================
tests/test_foo.py::test_foo
  test_foo.py:4: UserWarning: sync warning
    warnings.warn('sync warning')

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===================== 2 passed, 1 warning in 0.02s =====================

```

📋 **Your version of the Python**

```console
$ python --version
Python 3.7.3
```

📋 **Your version of the aiohttp/yarl/multidict distributions**

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.6.2
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: Nikolay Kim
Author-email: fafhrd91@gmail.com
License: Apache 2
Location: /home/vitja/work/testsuite/.venv-docs/lib/python3.7/site-packages
Requires: yarl, multidict, chardet, attrs, async-timeout
Required-by: testsuite, pytest-aiohttp
```
```console
$ python -m pip show multidict
Name: multidict
Version: 4.7.5
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /home/vitja/work/testsuite/.venv-docs/lib/python3.7/site-packages
Requires: 
Required-by: yarl, aiohttp
```
```console
$ python -m pip show yarl
Name: yarl
Version: 1.4.2
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /home/vitja/work/testsuite/.venv-docs/lib/python3.7/site-packages
Requires: idna, multidict
Required-by: aiohttp
```

📋 **Additional context**

### PR DESCRIPTION
**Pull request by vitek**
## What do these changes do?

< NOT FILLED OUT >

## Are there changes in behavior for the user?

< NOT FILLED OUT >

## Related issue number

Fixes #4656

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: aiohttp/pytest_plugin.py [hunk 1]
```diff
@@ -129,6 +129,13 @@ def _runtime_warning_context():  # type: ignore
                 '\n'.join(rw)
             ))
 
+    # Propagate warnings to pytest
+    for msg in _warnings:
+        warnings.showwarning(
+            msg.message,   # type: ignore
```
Comment (vitek) [inline aiohttp/pytest_plugin.py:135]: Python supports passing `Warning` instance as well as string, mypy is not aware. We can fetch message from the `Warning` instance but this way some extra information stored within `Warning` may be lost.

File: tests/test_pytest_plugin.py [hunk 2]
```diff
@@ -243,3 +246,13 @@ def test_finalized() -> None:
     testdir.makeconftest(CONFTEST)
     result = testdir.runpytest('-p', 'no:sugar', '--aiohttp-loop=pyloop')
     result.assert_outcomes(passed=2)
+
+
+def test_warnings_propagated():
+    with warnings.catch_warnings(record=True) as log:
```
Comment (webknjaz) [inline tests/test_pytest_plugin.py]: You should use a built-in `recwarn` fixture for testing warnings in pytest: https://docs.pytest.org/en/latest/warnings.html#recwarn.

File: setup.cfg [hunk 3]
```diff
@@ -38,6 +38,7 @@ addopts = --cov=aiohttp -v -rxXs --durations 10
 filterwarnings =
     error
     ignore:module 'ssl' has no attribute 'OP_NO_COMPRESSION'. The Python interpreter is compiled against OpenSSL < 1.0.0. Ref. https.//docs.python.org/3/library/ssl.html#ssl.OP_NO_COMPRESSION:UserWarning
+    always:test warning is propagated::test_pytest_plugin
```
Comment (webknjaz) [inline setup.cfg]: This is probably unnecessary when the proper fixture is used.

File: CHANGES/4656.bugfix [hunk 4]
```diff
@@ -0,0 +1 @@
+Fix pytest warnings propagation
```
Comment (webknjaz) [inline CHANGES/4656.bugfix]:
```suggestion
Propagate all warnings captured in coroutine test functions to pytest.
```

File: tests/test_pytest_plugin.py (hunk 2 above)
Comment (vitek) [reply tests/test_pytest_plugin.py]: Fixed, thanks!

**Pull request merged**